polygeist/tools/cgeist/Test/Verification/ext_vector_type.cpp

```diff
@@ -12,13 +14,19 @@ size_t evt2() {
   return stv.x;
 }
 
-// CHECK:   func.func @_Z3evtv() -> i32 attributes {llvm.linkage = #llvm.linkage<external>} {
-// CHECK-NEXT:    %alloca = memref.alloca() : memref<1x3xi32>
-// CHECK-NEXT:    %0 = affine.load %alloca[0, 0] : memref<1x3xi32>
-// CHECK-NEXT:    return %0 : i32
+// CHECK:   func.func @_Z3evtv() -> i64 attributes {llvm.linkage = #llvm.linkage<external>} {
+// CHECK-NEXT:    %c0_i64 = arith.constant 0 : i64
+// CHECK-NEXT:    %alloca = memref.alloca() : memref<1xvector<3xi64>>
```

Comment:
I am concerned that the alloca for a `size_t_vec stv` is identical to the one that would be generated for `size_t vec_stv[1]`. I think in this case this line should alloca a `memref<vector<3xi64>`. IMO this is incorrect.

Reply:
It is true that array of size 1 is not generate correctly, but that is a generic issue with cgeist, e.g., a simple `size_t arr[1]` is not generate as array in cgeist.

Reply:
Do you disagree that representing `  size_t_vec stv;` as a memref<1xvector<3xi64>> is incorrect and that the correct representation should be ` memref<vector<3xi64>>` ? 

I think what you mean is that the current `cgeist` implementation is *already* incorrect and so we need to fix the problem in a separate PR (or set of PRs). If so I am fine with doing that separately. But I want to make sure we agree the code is not correct as it is shown here.

Reply:
Please open an issue if you want to address it later.